lib/zone/zone-impl.ts

```diff
@@ -184,4 +185,12 @@ export class ZoneImpl implements zone.Zone {
             transportContext: transportContext
         };
     }
+
+    public on(event: string, func: any) {
```

Comment:
>func: any [](start = 29, length = 9)

func: (...args: any[]) => any 

Use this to save the logic of checking type of func below?

Reply:
good suggestion, fixed.